### PR TITLE
update git-extras plugin

### PR DESCRIPTION
**Pull request by spacewander**
The git-extras plugin has not been updated for long, so I do it to catch up with the development of git-extras

### CODE REVIEW

**Comment by alx741** [comment]
+1

**Comment by KhasMek** [comment]
This commit has been working perfectly for me the last while. However, I do not use gitfast, nor have I tested for the interoperability of them.

**Comment by ncanceill** [comment]
Can we have a few more testers here?

**Comment by GuillaumeSeren** [comment]
Hello,
I had the same issue, (completion not up to date),
and started on my own, but would be great to have this upgrade.
I'll test it, and add my feedback, 
but from the diff I read it is good and implement a lot of missing stuff.

Guillaume

**Comment by GuillaumeSeren** [comment]
I've merged this patch in my branch,
it work's well, it would be great to get it pulled.

I also will have more command to upgrade after this PR if possible.

Guillaume.

**Comment by GuillaumeSeren** [comment]
Hello,
I have been using this PR for 1 week and it work well,
I am working on a more upgraded version and so it would be great that this PR be merged,
that way I can send my version in near future.

Guillaume.

File: plugins/git-extras/git-extras.plugin.zsh [hunk 1]
```diff
@@ -227,59 +270,62 @@ _git-refactor() {
...
+    archive-file:'export the current HEAD of the git repository to a archive' \
+    back:'undo and stage latest commits' \
+    bug:'create a bug branch' \
     changelog:'populate changelog file with commits since the previous tag' \
+    commits-since:'list commits since a given date' \
     contrib:'display author contributions' \
     count:'count commits' \
+    create-branch:'create local and remote branch' \
     delete-branch:'delete local and remote branch' \
+    delete-merged-brancees:'delete merged branches'\
```
Comment (nicolas-brousse) [inline plugins/git-extras/git-extras.plugin.zsh]: It looks there is a typo with `delete-merged-brancees`. I think it must be `delete-merged-branches`.

**Comment by mcornella** [comment]
Also are missing:
- `obliterate`
- `pull-request`
as @nicolas-brousse said in https://github.com/robbyrussell/oh-my-zsh/pull/3464#issuecomment-111859661.

**Comment by spacewander** [comment]
Well, @GuillaumeSeren is working on a new pr with the latest git-extras version.
I will close this pr once he has completed it.
Currently just hope to keep this one open so that we can have a place to discuss.

**Comment by GuillaumeSeren** [comment]
Hi there,
@spacewander you want me to push your commit in my PR ?

Guillaume.

**Comment by spacewander** [comment]
@GuillaumeSeren 
Of course not, I mean I have abandoned this pr because your one will be better.

**Comment by GuillaumeSeren** [comment]
@spacewander  ok but I'll keep my work on top of your's.

So the commit itself still exist.

**Comment by mcornella** [comment]
I think this is good enough to merge.
:+1: 
/cc @robbyrussell

**Comment by GuillaumeSeren** [comment]
Great news !

As I previously say I support this PR to be merged because I based my work on it.

I hope next upgrade of git-extras can be merged faster because it end by punishing the oh-my-zsh's user while a bash user get the fix by upstream (git-extras).

:+1:

**Comment by benjaoming** [comment]
Thanks for updating git-extras! I was wondering if one of you would have the time to look at this bug, because apparently reverting this PR fixes the issue:

*Git completion partly broken #4395 *

**Comment by apjanke** [comment]
Hey folks, what do you think about seeing if upstream will take over maintenance of the `zsh` completions? Seems like there's interest: tj/git-extras#376.

**Comment by GuillaumeSeren** [comment]
@apjanke 
I agree for some project like that where the completion is evolving fast and quite complex,
it would be better/faster to keep the completion file in the upstream project.
Ideally the file can be added aside the bash completion file in /etc .

**Comment by apjanke** [comment]
Cool.

For zsh, it would go under `share/zsh/site-functions` or something similar, which is where zsh conventionally keeps its completion definitions, not under `/etc`.

**Comment by GuillaumeSeren** [comment]
Quoting Andrew Janke (2015-10-02 12:40:33)
> For zsh, it would go under share/zsh/site-functions or something similar, which
> is where zsh conventionally keeps its completion definitions, not under /etc.
Yes in deed,
the Makefile of git-extras can be upgraded to put the completion file in
'/usr/share/zsh/'.
We need to add a check in the git-extras zsh plugin to test if the install got
this file in '/usr/share/zsh' not a big deal.
I am working on a backport of the completion file in git-extras project,
after that it will be easier to keep it up-to-date.

**Pull request merged**